plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1769,6 +1769,10 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
         else {
             Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
 
+            Set<Integer> partitionSpecIds = icebergTable.snapshot(table.getSnapshotId().orElseThrow()).allManifests().stream()
```

Comment:
call me ignorant, but under what conditions would table.getSnapshotId() be empty?

Reply:
Good question. It's only ever empty during new table creation